pkg/prometheus/operator.go

```diff
@@ -2028,3 +2037,26 @@ func (c *Operator) listMatchingNamespaces(selector labels.Selector) ([]string, e
 	}
 	return ns, nil
 }
+
+// validateRemoteWriteSpec checks that mutually exclusive configurations are not
+// included in the Prometheus' remoteWrite configuration section.
+// Reference:
+// https://github.com/prometheus/prometheus/blob/main/docs/configuration/configuration.md#remote_write
+func validateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
+	mxFields := []interface{}{
+		spec.BasicAuth,
+		spec.OAuth2,
+		spec.Authorization,
+		spec.Sigv4}
+	var count int
+	for i, v := range mxFields {
+		if !reflect.ValueOf(v).IsNil() {
+			count++
+		}
+		if count > 1 {
+			return fmt.Errorf("invalid RemoteWriteSpec:\n%#v is not allowed together with\n%#v",
+				mxFields[i], mxFields[i-1])
```

Comment:
this might not report the fields in conflict: if `BasicAuth` and `Sigv4` are both defined, it will log `Authorization` and `Sigv4`.